saltstack/base/salt/prerequisites/usr/bin/user-data-helper.sh

```diff
@@ -1,6 +1,19 @@
 #!/bin/bash
 
 set -ex
+
+ACC_KEY_TMP_FILE=$(mktemp)
+chmod 600 "$ACC_KEY_TMP_FILE"
+trap 'rm -f "$ACC_KEY_TMP_FILE"' EXIT
+
+cat > ${ACC_KEY_TMP_FILE} <<EOF
+CCM_V2_AGENT_ACCESS_KEY_ID=$CCM_V2_AGENT_ACCESS_KEY_ID
+CCM_V2_AGENT_ENCIPHERED_ACCESS_KEY=$CCM_V2_AGENT_ENCIPHERED_ACCESS_KEY
+EOF
+
+unset CCM_V2_AGENT_ACCESS_KEY_ID
+unset CCM_V2_AGENT_ENCIPHERED_ACCESS_KEY
```

Comment:
In general, I suggest moving this block (or what remains from it :smile: ) inside `ccmv2-helper.sh`. The environment variables `CCM_V2_AGENT_ACCESS_KEY_ID` and `CCM_V2_AGENT_ENCIPHERED_ACCESS_KEY` are also visible there (and so are the other CCMv1, HTTP Proxy, Salt etc. ones that all come from the EC2 user data or the AWS Secrets Manager secret).